### PR TITLE
Document building and running Vespa C++ code using sanitizers

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,6 +197,77 @@ Note that the system test scrips are already in your PATH inside the Docker cont
...
+* `address` - instrument using [AddressSanitizer](https://github.com/google/sanitizers/wiki/AddressSanitizer)
+* `thread` - instrument using [ThreadSanitizer](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual)
+
+It's not possible to instrument with multiple sanitizers at the same time.
+
+Example using the `bootstrap-cmake.sh` script found in the Vespa root directory to build
+Vespa instrumented with ThreadSanitizer, running as an unprivileged user and building in-tree:
+
+```
+./bootstrap-cmake.sh -u . -DVESPA_USE_SANITIZER=thread
```
Comment (toregge) [inline README.md:218]: Consider just `cmake3 -DVESPA_USE_SANITIZER=thread .`

File: README.md [hunk 2]
```diff
@@ -197,6 +197,77 @@ Note that the system test scrips are already in your PATH inside the Docker cont
...
+
+If processes emit fatal sanitizer warnings on startup, e.g:
+
+```
+==51385==FATAL: ThreadSanitizer: failed to intercept munmap
+```
+
+then this is usually a sign that there are traces of a previous (non-instrumented) vespamalloc
+build in your Vespa install tree. Vespa startup scripts will implicitly pick up and load
+Vespamalloc if it's present, regardless of instrumentation status. The easiest way to get
```
Comment (geirst) [inline README.md:268]: Nitpick: Just use 'vespamalloc' (lowercase) as done elsewhere?

**Review by geirst — APPROVED** [review]
👍

File: README.md (hunk 2 above)
Comment (vekterli) [reply README.md]: Done

**Pull request merged**